fth/file.fth

```diff
@@ -127,4 +127,18 @@ create (LINE-TERMINATOR) \n c,
     ENDCASE
 ; immediate
 
+\ We basically try to open the file in read-only mode.  That seems to
+\ be the best that we can do with ANSI C.  If we ever want to do
+\ something more sophisticated, like calling access(2), we must create
+\ a proper primitive.  (OTOH, portable programs can't assume much
+\ about FILE-STATUS and non-portable programs could create a custom
+\ function for access(2).)
+: FILE-STATUS ( c-addr u -- x ior )
+    r/o bin open-file           ( fileid ior1 )
+    ?dup
+    IF                          ( fileid ior1 )
```

Comment:
fileid is undefined in the failure case.
if we want to provide implementation defined file information in the future then it would be better to return x=0 in both case. Then in the future x<>0 to point to some info.